### DISCUSSION TITLE
How to all the events including full day events, hourly events and recuring events?

### DISCUSSION THREAD

**Discussion by meet1919** [kuzmoyev/google-calendar-simple-api#141] [Q&A]
```python
from gcsa.google_calendar import GoogleCalendar
from google.oauth2 import service_account
import pytz
from datetime import datetime

start_date_cal = datetime.strptime('2022-08-01 00:00:00', '%Y-%m-%d %H:%M:%S')
end_date_cal = datetime.strptime('2022-09-01 00:00:00', '%Y-%m-%d %H:%M:%S')

SCOPES = ['https://www.googleapis.com/auth/calendar']
SERVICE_ACCOUNT_FILE = 'client_secrets2.json'

CAL_ID = 'xyz@gmail.com'

# authorize
credentials = service_account.Credentials.from_service_account_file(SERVICE_ACCOUNT_FILE, scopes=SCOPES)
calendar_auth = GoogleCalendar(CAL_ID, credentials=credentials)

calendar_events = calendar_auth.get_events(start_date_cal, end_date_cal, timezone='Israel')

```

I have a user of Israel and I want to fetch his events in google calendar. The code above fetches only 5 events which are full day events. There are around 35-40 events but I am getting very few. What am I missing?

I tried adding maxResults argument but I am getting error 
```
TypeError: Got an unexpected keyword argument kwargs
```
```python
calendar_events = calendar_auth.get_events(start_date_cal, end_date_cal, timezone='Israel', kwargs={'maxResults': 2500})
```

**Comment by kuzmoyev** [comment]
1. To pass an additional API parameter(s) you just pass them as a normal argument:
```
calendar_auth.get_events(..., maxResults=2500)
```
or unpack the dictionary
```
calendar_auth.get_events(..., **{'maxResults': 2500})
```

2. `maxResults` parameter is the size of the single response page. GCSA will go through all the pages automatically, so all the events will be fetched regardless of the value of `maxResults`.

4. `get_events` returns all the events including full-day, hourly, and recurring events. But it only returns the main event for the recurring events. To return all the instances of the recurring event you should specify `single_events` parameter:
```
calendar_auth.get_events(start_date_cal, end_date_cal, timezone='Israel', single_events=True)
```

**Reply by meet1919** [reply]
I used `single_events=True` and got 5 results from august month, without it I got 2. But I should get minimum 35 events. Why is this happening? The from the calendar settings I shared the calendar to my service account and set the permission to see all the events. What is the problem?

![Screenshot (70)](https://user-images.githubusercontent.com/62868163/188062175-3e317dc7-f301-4a76-abae-0d6e9bdc70a8.png)

**Reply by kuzmoyev** [reply]
How are those events that aren't retrieved different from those you've got?
You can try `. get_events(..., showDeleted=True)` option. Otherwise, I don't really know what might cause the issue.

**Reply by meet1919** [reply]
There are full-day events and hourly events. That's it. `showDeleted=True` got 9 more events then earlier but they arent relevant.

**Reply by kuzmoyev** [reply]
Could be that events' visibility is set to "private"? Can you access them from your own (non-service) account?

**Reply by meet1919** [reply]
Do you mean by `oauth 2.0` ?

**Reply by kuzmoyev** [reply]
Yes. If you used your own account instead of the service account

**Reply by meet1919** [reply]
I tried but I got insufficient scopes. To see all the event's titles, descriptions, and timezone, it required adding a sensitive scope which I wasn't able to add because the GCP app is a testing app.

**Reply by meet1919** [reply]
I haven't enabled domain-wide delegation for the service account which I am using. Will this be causing the error?
